parsers/mainz.py

```diff
@@ -0,0 +1,177 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup as parse
+import re
+import datetime
+import os.path as path
+
+from utils import Parser
+from pyopenmensa.feed import OpenMensaCanteen
+
+
+day_regex = re.compile('(\d{2}.\s\w+\s\d{4})')
+price_regex = re.compile('(?P<price>\d+[,.]\d{2}) ?€')
+notes_regex = re.compile('\[(?:(([A-Za-z0-9]+),?)+)\]$')
+extract_legend = re.compile('\((\w+,?)+\)')
+extract_legend_notes = re.compile('(?<=[\(,])(\w{1,2})')
+
+canteens = {
+  # API Extraction: https://github.com/kreativmonkey/jgu-mainz-openmensa/issues/1
+  '0' : 'all',
+  '1' : 'zentralmensa',
+  '2' : 'mensa-georg-foster',
+  '3' : 'cafe-rewi',
+  '4' : 'mensa-bingen',
+  '5' : 'mensa-K3',
+  '6' : 'mensa-holzstraße',
+  '7' : 'mensarium',
+  '8' : 'cafe-bingen-rochusberg',
+  '9' : 'mensablitz'
+}
+
+display = {
+  '2' : 'Aktuelle Woche',
+  '3' : 'Nächste Woche'
+}
+
+roles = ('student', 'other', 'employee')
+
+extraLegend = {
+    # Source: https://www.studierendenwerk-mainz.de/essentrinken/speiseplan/
+    '1': 'mit Farbstoff',
+    '2': 'mit Konservierungsstoff',
+    '3': 'mit Antioxidationsmittel',
+    '4': 'mit Geschmacksverstärker',
+    '5': 'geschwefelt',
+    '6': 'geschwärzt',
+    '7': 'gewachst',
+    '8': 'Phosphat',
+    '9': 'mit Süßungsmitteln',
+    '10': 'enthält eine Phenylalaninquelle',
+    'S' : 'Schweinefleisch',
+    'G' : 'Geflügelfleisch',
+    'R' : 'Rindfleisch',
+    'Gl' : 'Gluten',
+    'We' : 'Weizen (inkl. Dinkel)',
+    'Ro' : 'Roggen',
+    'Ge' : 'Gerste',
+    'Haf': 'Hafer',
+    'Kr' : 'Krebstiere und Krebstiererzeugnisse',
+    'Ei' : 'Eier und Eiererzeugnisse',
+    'Fi' : 'Fisch und Fischerzeugnisse',
+    'En' : 'Erdnüsse und Erdnusserzeugnisse',
+    'So' : 'Soja und Sojaerzeugnisse',
+    'La' : 'Milch und Milcherzeugnisse',
+    'Sl' : 'Sellerie und Sellerieerzeugnisse',
+    'Sf' : 'Senf und Senferzeugnisse',
+    'Se' : 'Sesamsamen und Sesamsamenerzeugnisse',
+    'Sw' : 'Schwefeldioxid und Sulfite > 10mg/kg',
+    'Lu' : 'Lupine und Lupinerzeugnisse',
+    'Wt' : 'Weichtiere und Weichtiererzeugnisse',
+    'Nu' : 'Schalenfrüchte',
+    'Man': 'Mandel',
+    'Has': 'Haselnüsse',
+    'Wa' : 'Walnüsse',
+    'Ka' : 'Kaschunüsse',
+    'Pe' : 'Pecanüsse',
+    'Pa' : 'Paranüsse',
+    'Pi' : 'Pistatien',
+    'Mac': 'Macadamianüsse',
+    'icon:S.png' : 'Scheinefleisch',
```

Comment:
Consider splitting these off into their own dictionary, e.g. `iconLegend`, to avoid the `icon:` prefix. See line 112 for how this can simplify code.